File: src/main/java/aginsun/kingdoms/server/ConfigTemplate.java
package aginsun.kingdoms.server;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public final class ConfigTemplate
{
    private HashMap<String, Integer> prices = new HashMap<>(), sales = new HashMap<>();

    public void setPrices(String name, int price)
    {
        prices.put(name, price);
    }

    public void setSales(String name, int price)
    {
        sales.put(name, price);
    }

    public int getPrices(String name)
    {
        return prices.get(name) != null ? prices.get(name) : 0;
    }

    public int getSales(String name)
    {
        return sales.get(name) != null ? prices.get(name) : 0;
    }

    public List<String> getNames()
    {
        return new ArrayList<>(prices.keySet());
    }

    public List<String> getSalesName()
    {
        return new ArrayList<>(sales.keySet());
    }
}

File: src/main/java/kingdoms/server/ConfigTemplate.kt
package kingdoms.server

import com.google.gson.annotations.SerializedName
import java.util.*

internal class ConfigTemplate
{
    @SerializedName("onlyOpBuild")
    var isOp: Boolean = false

    private val prices = HashMap<String, Int>()
    private val sales = HashMap<String, Int>()

    val names: List<String>
        get() = ArrayList(prices.keys)

    val salesName: List<String>
        get() = ArrayList(sales.keys)

    fun setPrices(name: String, price: Int)
    {
        prices[name] = price
    }

    fun setSales(name: String, price: Int)
    {
        sales[name] = price
    }

    fun getPrices(name: String): Int?
    {
        return if (prices[name] != null) prices[name] else 0
    }

    fun getSales(name: String): Int?
    {
        return if (sales[name] != null) prices[name] else 0
    }
}